openedx/adg/lms/webinars/api_views.py

```diff
@@ -0,0 +1,60 @@
+"""
+API views for webinar app
+"""
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponseServerError
+from django.utils.decorators import method_decorator
+from django.utils.translation import ugettext as _
+from django.views.decorators.csrf import ensure_csrf_cookie
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK
+from rest_framework.views import APIView
+
+from openedx.core.lib.api.view_utils import view_auth_classes
+
+from .helpers import send_webinar_registration_email
+from .models import Webinar, WebinarRegistration
+
+
+@view_auth_classes(is_authenticated=True)
+class WebinarRegistrationView(APIView):
+    """
+    A view to register user in a webinar event or cancel already registered user in a webinar.
+    """
```

Comment:
We can make use of [UpdateAPIView](https://www.django-rest-framework.org/api-guide/generic-views/#updateapiview) to make this cleaner.

Just a suggestion!

Reply:
Also take a look at [UpdateModelMixin](https://www.django-rest-framework.org/api-guide/generic-views/#updatemodelmixin).